x/fee-delegation/keeper.go

```diff
@@ -0,0 +1,82 @@
+package fee_delegation
+
+import (
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      *codec.Codec
+}
+
+func NewKeeper(storeKey sdk.StoreKey, cdc *codec.Codec) Keeper {
+	return Keeper{storeKey, cdc}
+}
+
+func FeeAllowanceKey(grantee sdk.AccAddress, granter sdk.AccAddress) []byte {
+	return []byte(fmt.Sprintf("f/%x/%x", grantee, granter))
+}
+
+func (k Keeper) DelegateFeeAllowance(ctx sdk.Context, grantee sdk.AccAddress, granter sdk.AccAddress, allowance FeeAllowance) {
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshalBinaryBare(allowance)
+	store.Set(FeeAllowanceKey(grantee, granter), bz)
+}
+
+func (k Keeper) RevokeFeeAllowance(ctx sdk.Context, grantee sdk.AccAddress, granter sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(FeeAllowanceKey(grantee, granter))
+}
+
+type FeeAllowanceGrant struct {
+	Allowance FeeAllowance   `json:"allowance"`
+	Grantee   sdk.AccAddress `json:"grantee"`
+	Granter   sdk.AccAddress `json:"granter"`
+}
+
+func (k Keeper) GetFeeAllowances(ctx sdk.Context, grantee sdk.AccAddress) []FeeAllowanceGrant {
```

Comment:
The pattern we've been following in the SDK is to instead provide/expose iterators with a cb function param instead. We can keep this method, but we should also have an iterator method which this would simply use.